Jenkinsfile

```diff
@@ -274,21 +249,33 @@ def getParallelTests(String image) {
     return ret;
 }
 
-node {
-
-    def isDefaultBranch = (env.BRANCH_NAME == 'master') 
-    def daysToKeep = '20';
-    def numToKeep = (isDefaultBranch ? '-1' : '10');
-
-    properties([
-        buildDiscarder(logRotator(
-            daysToKeepStr:         daysToKeep,
-            artifactDaysToKeepStr: daysToKeep,
+def bazelTest = {
+    stage ('Build with Bazel') {
+        node {
+            stage('Setup Bazel Build') {
+                checkout scm;
+                sh label: 'Setup Docker Image', script: 'docker build -f docker/Dockerfile.bazel -t openroad/bazel-ci .';
+            }
+            withDockerContainer(args: '-u root -v /var/run/docker.sock:/var/run/docker.sock', image: 'openroad/bazel-ci:latest') {
+                stage('Bazel Build') {
+                    withCredentials([file(credentialsId: 'bazel-cache-sa', variable: 'GCS_SA_KEY')]) {
+                        timeout(time: 120, unit: 'MINUTES') {
+                            def cmd = 'bazelisk test --config=ci --show_timestamps --test_output=errors --curses=no --force_pic';
+                            if (env.BRANCH_NAME != 'master') {
+                                cmd += ' --remote_upload_local_results=false';
+                            }
+                            sh label: 'Bazel Build', script: cmd + ' --google_credentials=$GCS_SA_KEY ...';
+                        }
+                        sh label: 'Archive logs', script:  'tar zcvf bazel-logs.tgz bazel-testlogs/.';
+                        archiveArtifacts artifacts: 'bazel-logs.tgz';
```

Comment:
@oharboe 
Does this address https://github.com/The-OpenROAD-Project/OpenROAD/pull/7722#issuecomment-3042332591, or do we need to retain more files/logs from elsewhere?